packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2185,6 +2185,18 @@ abstract class RenderObject with DiagnosticableTreeMixin implements HitTestTarge
   }
   bool _needsLayout = true;
 
+  /// The nearest relayout boundary enclosing this render object, if known.
+  ///
+  /// When not null, the relayout boundary is either this render object itself
+  /// or one of its ancestors, and all the render objects in the ancestry chain
+  /// up through that ancestor have the same [_relayoutBoundary].
+  /// Equivalently: when not null, the relayout boundary is either this render
+  /// object itself or the same as that of its parent.  (So [_relayoutBoundary]
+  /// is one of `null`, `this`, or `parent!._relayoutBoundary!`.)
+  ///
+  /// This is set in [layout], and consulted by [markNeedsLayout] in deciding
+  /// whether to recursively mark the parent as also needing layout.
+  // TODO(gnprice): Just when is _relayoutBoundary null, and what does that mean?
```

Comment:
I believe null just means the relayout boundary is currently not known. This can happen when the RO is currently not part of the tree. During layout, every RO will be assigned a relayout boundary, though.

Reply:
Thanks, this was a good prompt for me to go try to take care of this TODO now. I've just pushed an added commit that replaces it with this paragraph:
```dart
  /// This property is initially null, and becomes null again if this
  /// render object is removed from the tree (with [dropChild]);
  /// it remains null until the first layout of this render object
  /// after it was most recently added to the tree.
  /// This property can also be null while an ancestor in the tree is
  /// currently doing layout, until this render object itself does layout.
```

I initially wrote just the first sentence, which is a way of saying much the same thing as your comment, because that sounded right to me. Then I figured I should fact-check it to make sure, so searched again through the places the property gets set to null… and was reminded there's one more wrinkle to it, namely the `_visitChildren(_cleanChildRelayoutBoundary)` call in `layout`. (Which must have been part of why I left this as a TODO in the first place.)

But then I added the second sentence, and I think that covers it well enough to cut the TODO.

That particular wrinkle is one that will go away later in my series for #150524, so that'll happily make this a bit cleaner to explain.